src/core/bonded_interactions/angle_cosine.hpp

```diff
@@ -31,166 +31,101 @@
-                                           Particle *p_right,
-                                           Bonded_ia_parameters *iaparams,
-                                           double force1[3], double force2[3],
-                                           double force3[3]) {
-
-  int j;
-  double pot_dep;
-  double cos_phi;
-  double sin_phi;
-  double vec31[3];
-  double vec21[3];
-  double vec12[3]; // espresso convention
-  double vec21_sqr;
-  double vec31_sqr;
-  double vec21_magn;
-  double vec31_magn;
-  double fj[3];
-  double fk[3];
-  double fac;
-
-  get_mi_vector(vec12, p_mid->r.p, p_left->r.p);
-  for (j = 0; j < 3; j++)
-    vec21[j] = -vec12[j];
-
-  get_mi_vector(vec31, p_right->r.p, p_mid->r.p);
-  vec21_sqr = sqrlen(vec21);
-  vec21_magn = sqrt(vec21_sqr);
-  vec31_sqr = sqrlen(vec31);
-  vec31_magn = sqrt(vec31_sqr);
-  cos_phi = scalar(vec21, vec31) / (vec21_magn * vec31_magn);
-  sin_phi = sqrt(1.0 - Utils::sqr(cos_phi));
-
-  /* uncomment this block if interested in the angle
-  if(cos_phi < -1.0) cos_phi = -TINY_COS_VALUE;
-  if(cos_phi >  1.0) cos_phi =  TINY_COS_VALUE;
-  phi = acos(cos_phi);
-  */
-
-  {
-    double K, sin_phi0, cos_phi0;
-    K = iaparams->p.angle_cosine.bend;
-    sin_phi0 = iaparams->p.angle_cosine.sin_phi0;
-    cos_phi0 = iaparams->p.angle_cosine.cos_phi0;
-
+inline void calc_angle_cosine_3body_forces(Particle const *p_mid,
+                                           Particle const *p_left,
+                                           Particle const *p_right,
+                                           Bonded_ia_parameters const *iaparams,
+                                           Vector3d &force1, Vector3d &force2,
+                                           Vector3d &force3) {
+
+  auto forceFactor = [&iaparams](double &cos_phi, double &sin_phi) {
+    /* uncomment this block if interested in the angle
+    if(cos_phi < -1.0) cos_phi = -TINY_COS_VALUE;
+    if(cos_phi >  1.0) cos_phi =  TINY_COS_VALUE;
+    phi = acos(cos_phi);
+    */
+    auto K = iaparams->p.angle_cosine.bend;
+    auto sin_phi0 = iaparams->p.angle_cosine.sin_phi0;
+    auto cos_phi0 = iaparams->p.angle_cosine.cos_phi0;
     // potential dependent term [dU/dphi = K * sin(phi - phi0)]
     // trig identity: sin(a - b) = sin(a)cos(b) - cos(a)sin(b)
-    pot_dep = K * (sin_phi * cos_phi0 - cos_phi * sin_phi0);
-  }
-
-  fac = pot_dep / sin_phi;
-
-  for (j = 0; j < 3; j++) {
-    fj[j] =
-        vec31[j] / (vec21_magn * vec31_magn) - cos_phi * vec21[j] / vec21_sqr;
-    fk[j] =
-        vec21[j] / (vec21_magn * vec31_magn) - cos_phi * vec31[j] / vec31_sqr;
-  }
-
-  // note that F1 = -(F2 + F3)
-  for (j = 0; j < 3; j++) {
-    force1[j] = force1[j] - fac * (fj[j] + fk[j]);
-    force2[j] = force2[j] + fac * fj[j];
-    force3[j] = force3[j] + fac * fk[j];
-  }
+    auto pot_dep = K * (sin_phi * cos_phi0 - cos_phi * sin_phi0);
+    auto fac = pot_dep / sin_phi;
```

Comment:
`return`

Reply:
I'm returning a temporary variable named `fac` to follow the naming convention in other `forceFactor` lambdas. Should I avoid that?

Reply:
The lambdas now use a `return` directly.